File: frontend/src/services/unifiedSearchService.ts
import type { AssetPage } from '../types/asset'
import type { DocumentPage } from '../types/document'

const apiBaseUrl = import.meta.env.VITE_API_BASE_URL ?? ''

export interface UnifiedSearchParams {
  query: string
  platformFamily?: string
  platformVariant?: string
  productLine?: string
  base?: string
  productionLine?: string
  processSection?: string
  assetPage: number
  assetPerPage: number
  documentPage: number
  documentPerPage: number
}

export interface UnifiedSearchResponse {
  assets: AssetPage & { status: string; errorCode: string }
  documents: DocumentPage & { status: string; errorCode: string }
}

export async function searchUnified(params: UnifiedSearchParams): Promise<UnifiedSearchResponse> {
  const query = new URLSearchParams({
    q: params.query,
    asset_page: String(params.assetPage),
    asset_per_page: String(params.assetPerPage),
    document_page: String(params.documentPage),
    document_per_page: String(params.documentPerPage),
  })
  if (params.platformFamily) query.set('platform_family', params.platformFamily)
  if (params.platformVariant) query.set('platform_variant', params.platformVariant)
  if (params.productLine) query.set('product_line', params.productLine)
  if (params.base) query.set('base', params.base)
  if (params.productionLine) query.set('production_line', params.productionLine)
  if (params.processSection) query.set('process_section', params.processSection)
  const response = await fetch(`${apiBaseUrl}/api/v1/search?${query.toString()}`, { headers: { Accept: 'application/json' } })
  if (!response.ok) throw new Error(`统一检索失败：${response.status}`)
  return response.json() as Promise<UnifiedSearchResponse>
}

File: backend/src/main/java/com/tianshu/assets/search/application/UnifiedSearchService.java
package com.tianshu.assets.search.application;

import com.tianshu.assets.asset.application.AssetQueryService;
import com.tianshu.assets.asset.domain.AssetSearchCriteria;
import com.tianshu.assets.document.application.DocumentQueryService;
import com.tianshu.assets.document.domain.DocumentSearchCriteria;
import com.tianshu.assets.search.api.UnifiedSearchResponse;
import org.springframework.stereotype.Service;

@Service
public class UnifiedSearchService {

    private final AssetQueryService assets;
    private final DocumentQueryService documents;

    public UnifiedSearchService(AssetQueryService assets, DocumentQueryService documents) {
        this.assets = assets;
        this.documents = documents;
    }

    public UnifiedSearchResponse search(Criteria criteria) {
        var assetPage = assets.search(new AssetSearchCriteria(criteria.query(), null, null, "",
                criteria.platformFamily(), criteria.platformVariant(), criteria.baseName(), criteria.productionLine(),
                null, criteria.assetPage(), criteria.assetPerPage(), criteria.productLine(), criteria.processSection()));
        var documentPage = documents.search(new DocumentSearchCriteria(criteria.query(), "", criteria.platformFamily(),
                criteria.platformVariant(), criteria.productLine(), criteria.baseName(), criteria.productionLine(),
                criteria.processSection(), criteria.documentPage(), criteria.documentPerPage()));
        return UnifiedSearchResponse.from(assetPage, documentPage);
    }

    public record Criteria(String query, String platformFamily, String platformVariant, String productLine,
            String baseName, String productionLine, String processSection, int assetPage, int assetPerPage,
            int documentPage, int documentPerPage) {
        public Criteria {
            query = text(query);
            platformFamily = text(platformFamily);
            platformVariant = text(platformVariant);
            productLine = text(productLine);
            baseName = text(baseName);
            productionLine = text(productionLine);
            processSection = text(processSection);
            assetPage = Math.max(assetPage, 1);
            documentPage = Math.max(documentPage, 1);
            assetPerPage = Math.clamp(assetPerPage, 1, 100);
            documentPerPage = Math.clamp(documentPerPage, 1, 100);
        }

        private static String text(String value) { return value == null ? "" : value.trim(); }
    }
}
